byzcoin/service.go

```diff
@@ -504,11 +506,31 @@ func (s *Service) updateCollectionCallback(sbID skipchain.SkipBlockID) error {
 			"programmer error if you see this message.")
 	}
 
-	cdb := s.getCollection(sb.SkipChainID())
-	collectionIndex := cdb.getIndex()
+	// If we are the genesis block, create the trie.
+	if sb.Index == 0 {
+		var body DataBody
+		err := protobuf.DecodeWithConstructors(sb.Payload, &body, network.DefaultConstructors(cothority.Suite))
+		if err != nil {
+			log.Error(s.ServerIdentity(), "could not unmarshal body for genesis block", err)
+			return errors.New("couldn't unmarshal body for genesis block")
+		}
+		nonce := body.TxResults[0].ClientTransaction.Instructions[0].Spawn.Args.Search("trie_nonce")
+		if len(nonce) == 0 {
+			return errors.New("nonce is empty")
+		}
+		s.createStateTrie(sb.SkipChainID(), nonce)
```

Comment:
this was intentional actually, because afterwards we use `getStateTrie`. if something goes wrong then an error is returned. i added a comment to explain why the returned value isn't used